auditbeat/auditbeat.reference.yml

```diff
@@ -148,6 +148,50 @@ auditbeat.modules:
     # if the number of events stored in the queue is < min_flush_events.
     #flush.timeout: 1s
 
+  # The spool queue will store events in a local spool file, before
+  # forwarding the events to the outputs.
+  # The spool file is a circular buffer, which blocks once the file/buffer is full.
+  # Events are put into a write buffer and flushed once the write buffer
+  # is full or the flush_timeout is triggered.
+  # Once ACKed by the output, events are removed immediately from the queue,
+  # making space for new events to be persisted.
+  #spool:
```

Comment:
Why not call this `file` or `disk`? It would make it more obvious that is persistent.

Reply:
@ph @ruflin 

Hmmm... introducing `queue.type` would be a bc-breaking change. As one can have only one queue type, the queue config follows the dictionary style config-pattern like:

```
queue.<type>:
```

What wrong with `spool`?

Right now this one implements a FIFO. For metricbeat use cases we might actually introduce another persistent LIFO queue type. This one should have another type. How would you name any of these now?

Reply:
I don't think we should introduce a type because of the reasons mentioned by @urso. There can only be one queue at the time. 

For me spooling can be to disk or memory. Historically we had a spooler in filebeat which kept the data in memory. Other options would be to call it `file`?

+1 on the proposal from @ph about the priority. FIFO or LIFO is a config option of the queue. It can mean in the background a completely different implementation but the user should not have to worry about that.

Reply:
Regarding priority I was initially thinking the same. But if we have a priority setting, user must be allowed to change the setting between restarts.

But async ACK behaviour of the queue makes the on-disk structure a little more complicated. When introducing stack like functionality, we will end up with a many holes. That is, freeing space will be somewhat more complicated in the LIFO case. I'd like to solve the LIFO case separately, potentially merging both cases into a common file format, later in time. Priority based queue, using heaps might become even more complicated.

Reply:
@urso Look like a good compromise.

@ruflin You have a point, Disk or File at the discretion of @urso.